File: go/law_types.go
package legalize

import (
	"context"
	"net/http"
)

// LawTypesService exposes /api/v1/{country}/law-types.
type LawTypesService struct {
	client *Client
}

// List returns the law-type codes defined for a country (e.g. "ley",
// "real_decreto", "constitucion").
func (s *LawTypesService) List(ctx context.Context, country string) ([]string, error) {
	var out []string
	if err := s.client.requestJSON(ctx, http.MethodGet, API+"/"+country+"/law-types", nil, &out); err != nil {
		return nil, err
	}
	return out, nil
}

File: node/src/resources/lawTypes.ts
/** `/api/v1/{country}/law-types` — law types per country. */

import type { Legalize } from "../client.js";

const API = "/api/v1";

export class LawTypes {
  private readonly client: Legalize;
  constructor(client: Legalize) {
    this.client = client;
  }

  /** List law type identifiers (e.g. `["constitucion", "ley", "real_decreto"]`). */
  async list(country: string, options: { signal?: AbortSignal } = {}): Promise<string[]> {
    return this.client.request<string[]>("GET", `${API}/${country}/law-types`, {
      ...(options.signal ? { signal: options.signal } : {}),
    });
  }
}
